src/packages/auth/src/authentication/methods/utils.ts

```diff
@@ -67,3 +69,47 @@ export const runAfterHooks = async <D, H = CredentialInsertInput | CredentialUpd
 
 	return hookEntities;
 };
+
+export const makeLoginPasswordQuerySafeForLogging = (
+	queryString: string,
+	variables: VariableValues | undefined
+) => {
+	const safeVariables: VariableValues = JSON.parse(JSON.stringify(variables ?? {}));
+	const ast = parse(queryString);
+
+	const safeAst = visit(ast, {
+		enter(node) {
+			if (
+				node.kind === Kind.ARGUMENT &&
+				node.name.value === 'password' &&
+				node.value.kind === Kind.STRING
+			) {
+				return {
+					...node,
+					value: {
+						...node.value,
+						value: '********',
+					},
+				};
+			}
+
+			if (
+				node.kind === Kind.ARGUMENT &&
+				node.name.value === 'password' &&
+				node.value.kind === Kind.VARIABLE
+			) {
```

Comment:
Would be a little bit more readable I think to have one big `if`  with `node.kind === Kind.ARGUMENT && node.name.value === 'password'` then inside have the check for `node.value.kind`.

Reply:
Totally, I don't know how that didn't occur to me.